src/Provider/AbstractProvider.php

```diff
@@ -222,7 +217,7 @@ public function setHttpClient(HttpClientInterface $client)
     /**
      * Returns the HTTP client instance.
      *
-     * @return HttpClientInterface
+     * @return ClientInterface
      */
     public function getHttpClient()
```

Comment:
Is this method really needed? But it is maybe out of scope for this PR.

Reply:
Yeah, I'd keep it for now, but since this is a BC break, this will have to be 3.x anyway.